addons/sale/views/sale_portal_templates.xml

```diff
@@ -506,7 +488,7 @@
                                             <t t-esc="(1-line.discount / 100.0) * line.price_unit" t-options='{"widget": "monetary", "display_currency": sale_order.pricelist_id.currency_id}'/>
                                         </div>
                                     </td>
-                                    <td t-if="display_discount" class="text-left">
+                                    <td t-if="display_discount" t-attf-class="text-right {{ 'd-none d-sm-table-cell' if report_type == 'html' else '' }}">
```

Comment:
Hidding discount in small resolution is probably not a great idea.. This information is as much important as the others.